src/java.base/share/classes/java/io/File.java

```diff
@@ -2176,6 +2176,12 @@ public static File createTempFile(String prefix, String suffix,
             }
         } while (fs.hasBooleanAttributes(f, FileSystem.BA_EXISTS));
 
+
+        if(!tmpdir.exists() || !tmpdir.canWrite()) {
```

Comment:
Should this check that `tmpdir.isDirectory()` that will do both existence & directory check? That would catch if java.io.t odor was mistakenly set to an existing (writable) file.

```suggestion
        if(!tmpdir.isDirectory() || !tmpdir.canWrite()) {
```

Reply:
Using a new approach, please check the new commits. Thanks.